daemon/archive_tarcopyoptions_unix.go

```diff
@@ -12,14 +16,77 @@ func (daemon *Daemon) tarCopyOptions(container *container.Container, noOverwrite
 	if container.Config.User == "" {
 		return daemon.defaultTarCopyOptions(noOverwriteDirNonDir), nil
 	}
-
-	user, err := idtools.LookupUser(container.Config.User)
+	uid, gid, err := getUIDGID(container.Config.User)
 	if err != nil {
 		return nil, err
 	}
 
 	return &archive.TarOptions{
 		NoOverwriteDirNonDir: noOverwriteDirNonDir,
-		ChownOpts:            &idtools.IDPair{UID: user.Uid, GID: user.Gid},
+		ChownOpts:            &idtools.IDPair{UID: uid, GID: gid},
 	}, nil
 }
+
+// getUIDGID resolves the UID and GID of a given user and, optionally, group.
+//
+// usergrp is a username or uid, and optional group, in the format `user[:group]`.
+// Both `user` and `group` can be provided as an `uid` / `gid`, so the following
+// formats are supported:
+//
+//   username            - valid username from /etc/passwd
```

Comment:
ah, yes, perhaps I should change that; I started writing for this PR, but found https://github.com/moby/moby/blob/c8a2596d676a815cc17162f178869be4aeb4b301/daemon/daemon_unix.go#L941-L951, so copied parts from that

I'm not sure what approach should be taken though if the user needs to come from within the container (where `getent` may not be available, such as in a `busybox` container) - that's why I kept it as "WIP"